sway-lsp/src/server.rs

```diff
@@ -1302,19 +1264,59 @@ mod tests {
         )
         .await;
 
-        turbofish_definition_check_option(&mut service, &uri, 15, 12, 1).await;
-        turbofish_definition_check_option(&mut service, &uri, 16, 17, 2).await;
-        turbofish_definition_check_option(&mut service, &uri, 17, 29, 3).await;
-        turbofish_definition_check_option(&mut service, &uri, 18, 19, 4).await;
+        let mut opt_go_to = GotoDefintion {
+            req_uri: &uri,
+            req_line: 15,
+            req_char: 12,
+            def_line: 80,
+            def_start_char: 9,
+            def_end_char: 15,
+            def_path: "sway-lib-std/src/option.sw",
+        };
+
+        let _ = definition_check(&mut service, &opt_go_to, 1).await;
```

Comment:
nit to make the test cases more readable:

```
        fn get_go_to_test_case(
            base: GotoDefinition,
            req_line: u64,
            req_char: u64,
        ) -> GotoDefintion {
            GotoDefintion {
                ..base,
                req_line,
                req_char,
            }
        }

        fn get_opt_test_case(req_line: u64, req_char: u64) -> GotoDefintion {
            get_go_to_test_case(opt_go_to, req_line, req_char)
        }

        fn get_res_test_case(req_line: u64, req_char: u64) -> GotoDefintion {
            get_go_to_test_case(res_go_to, req_line, req_char)
        }

        vec![
            // option.sw
            get_opt_test_case(15, 12),
            get_opt_test_case(16, 17),
            get_opt_test_case(17, 29),
            get_opt_test_case(18, 19),
            get_opt_test_case(20, 13),
            get_opt_test_case(21, 19),
            get_opt_test_case(22, 29),
            get_opt_test_case(23, 18),
            // result.sw 
            get_res_test_case(20, 19),
            get_res_test_case(21, 25),
            get_res_test_case(22, 36),
            get_res_test_case(23, 27),
        ]
        .into_iter()
        .for_each(|(i, test_case)| {
            let _ = definition_check(&mut service, &go_to, i + 1).await;
        });
```

Reply:
Thanks, i've done something a little different but it looks a lot cleaner now.